src/Components/Endpoints/src/DependencyInjection/EndpointHtmlRenderer.cs

```diff
@@ -11,32 +13,30 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Extensions;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Components.Endpoints;
 
-// Wraps the public HtmlRenderer APIs so that the output also gets annotated with prerendering markers.
-// This allows the prerendered content to switch later into interactive mode.
-// This class also deals with initializing the standard component DI services once per request.
-internal sealed class ComponentPrerenderer : IComponentPrerenderer
+/// <summary>
+/// An <see cref="HtmlRendererCore"/> subclass that is used when prerendering on an endpoint
+/// or for the component tag helper. It knows how to annotate the output with prerendering
+/// markers so the content can later switch into interactive mode. It also deals with initializing
+/// the standard component DI services once per request.
+/// </summary>
+internal sealed class EndpointHtmlRenderer : HtmlRendererCore, IComponentPrerenderer
```

Comment:
Can we make the class partial and move the IComponentPrerender implementation into its own file? That way we keep this piece focused on the renderer logic itself

Reply:
Sure, good idea. I've split up `EndpointHtmlRenderer` into three partials now, representing (1) being a renderer, (2) prerendering components, and (3) prerendering state. It's a lot easier to read now.